types/src/chainspec/vm_config/messages_limits.rs

```diff
@@ -0,0 +1,113 @@
+#[cfg(feature = "datasize")]
+use datasize::DataSize;
+use rand::{distributions::Standard, prelude::*, Rng};
+use serde::{Deserialize, Serialize};
+
+use crate::bytesrepr::{self, FromBytes, ToBytes};
+
+/// Configuration for messages limits.
+#[derive(Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Debug)]
+#[cfg_attr(feature = "datasize", derive(DataSize))]
+#[serde(deny_unknown_fields)]
+pub struct MessagesLimits {
+    /// Maximum size (in bytes) of a topic name string.
+    pub max_topic_name_size: u32,
+    /// Maximum message size in bytes.
+    pub max_message_size: u32,
+    /// Maximum number of topics that a contract can register.
+    pub max_topics_per_contract: u32,
+}
+
+impl MessagesLimits {
+    /// Returns the max number of topics a contract can register.
+    pub fn max_topics_per_contract(&self) -> u32 {
+        self.max_topics_per_contract
+    }
+
+    /// Returns the maximum allowed size for the topic name string.
+    pub fn max_topic_name_size(&self) -> u32 {
+        self.max_topic_name_size
+    }
+
+    /// Returns the maximum allowed size (in bytes) of the serialized message payload.
+    pub fn max_message_size(&self) -> u32 {
+        self.max_message_size
+    }
+}
+
+impl Default for MessagesLimits {
+    fn default() -> Self {
+        Self {
+            max_topic_name_size: 256,
+            max_message_size: 1024,
+            max_topics_per_contract: 128,
+        }
+    }
+}
+
+impl ToBytes for MessagesLimits {
```

Comment:
Shall we add the serialization roundtrip test for `MessagesLimits`?

Reply:
I thought it was picked up by a serialization test for `WasmConfig` but it wasn't. Added here: https://github.com/casper-network/casper-node/pull/4315/commits/3190647ec0012448907208e3a8406244433c5486